File: API/UniUnboxdAPI/Models/NotificationSettings.cs
// This file contains the definition of the NotificationSettings class.
namespace UniUnboxdAPI.Models
{
    // The NotificationSettings class represents the notification settings for a student.
    public class NotificationSettings
    {
        // The ID of the student.
        public int StudentId { get; set; }

        // The student associated with the notification settings.
        public Student? Student { get; set; }

        // Indicates whether the student receives review emails from followers.
        public bool ReceivesFollowersReviewMail { get; set; } = true;

        // Indicates whether the student receives review push notifications from followers.
        public bool ReceivesFollowersReviewPush { get; set; } = true;

        // Indicates whether the student receives new reply emails.
        public bool ReceivesNewReplyMail { get; set; } = true;

        // Indicates whether the student receives new reply push notifications.
        public bool ReceivesNewReplyPush { get; set; } = true;

        // Indicates whether the student receives new follower emails.
        public bool ReceivesNewFollowerMail { get; set; } = true;

        // Indicates whether the student receives new follower push notifications.
        public bool ReceivesNewFollowerPush { get; set; } = true;
    }
}

File: Application/app/src/main/java/com/example/uniunboxd/models/student/NotificationSettings.java
package com.example.uniunboxd.models.student;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class NotificationSettings {
    public int StudentId;
    public boolean ReceivesFollowersReviewMail;
    public boolean ReceivesFollowersReviewPush;
    public boolean ReceivesNewReplyMail;
    public boolean ReceivesNewReplyPush;
    public boolean ReceivesNewFollowerMail;
    public boolean ReceivesNewFollowerPush;

    /**
     * Constructor for the NotificationSettings class.
     *
     * @param StudentId                The student's ID.
     * @param ReceivesFollowersReviewMail Whether the student receives followers' review mail.
     * @param ReceivesFollowersReviewPush Whether the student receives followers' review push.
     * @param ReceivesNewReplyMail       Whether the student receives new reply mail.
     * @param ReceivesNewReplyPush       Whether the student receives new reply push.
     * @param ReceivesNewFollowerMail    Whether the student receives new follower mail.
     * @param ReceivesNewFollowerPush    Whether the student receives new follower push.
     */
    @JsonCreator
    public NotificationSettings(@JsonProperty("studentId") int StudentId,
                                @JsonProperty("receivesFollowersReviewMail") boolean ReceivesFollowersReviewMail,
                                @JsonProperty("receivesFollowersReviewPush") boolean ReceivesFollowersReviewPush,
                                @JsonProperty("receivesNewReplyMail") boolean ReceivesNewReplyMail,
                                @JsonProperty("receivesNewReplyPush") boolean ReceivesNewReplyPush,
                                @JsonProperty("receivesNewFollowerMail") boolean ReceivesNewFollowerMail,
                                @JsonProperty("receivesNewFollowerPush") boolean ReceivesNewFollowerPush) {
        this.StudentId = StudentId;
        this.ReceivesFollowersReviewMail = ReceivesFollowersReviewMail;
        this.ReceivesFollowersReviewPush = ReceivesFollowersReviewPush;
        this.ReceivesNewReplyMail = ReceivesNewReplyMail;
        this.ReceivesNewReplyPush = ReceivesNewReplyPush;
        this.ReceivesNewFollowerMail = ReceivesNewFollowerMail;
        this.ReceivesNewFollowerPush = ReceivesNewFollowerPush;
    }
}

